server/src/main/java/org/opensearch/telemetry/tracing/channels/TraceableHttpChannel.java

```diff
@@ -49,7 +50,7 @@ private TraceableHttpChannel(HttpChannel delegate, Span span, Tracer tracer) {
      * @return http channel
      */
     public static HttpChannel create(HttpChannel delegate, Span span, Tracer tracer) {
-        if (FeatureFlags.isEnabled(FeatureFlags.TELEMETRY) == true) {
+        if (FeatureFlags.isEnabled(FeatureFlags.TELEMETRY) == true && TelemetrySettings.isTracerFeatureEnabled()) {
```

Comment:
I believe we should make a method in the `Tracer`:

```
boolean isRecording();
```

So to reflected the current state of the tracing feature (based on feature flag and telemetrySettings)